main.go

```diff
@@ -41,12 +48,21 @@ func EntryPrinter(cv chan bool, input chan *Entry) {
 			var newpath = strings.TrimLeft(strings.Replace(entry.path, pwd, "", 1), "/")
 			fmt.Printf("%s => %s", oldpath, newpath)
 		} else {
-			fmt.Printf("%s => %s\n", entry.path, entry.newpath)
+			fmt.Printf("%s => %s  .. %s\n", entry.path, entry.newpath, entry.result)
 		}
 	}
 	cv <- true
 }
 
+func GenerateFormatName() string {
+	c_format := strings.Replace(*replacementFormatPtr, "%i", "%04d", 1)
+	m.Lock()
+	ret_str := fmt.Sprintf(c_format, sequence_number)
+	sequence_number = sequence_number + 1
+	m.Unlock()
+	return ret_str
+}
+
```

Comment:
Actually we should pass a payload struct to the `GenerateReplaceFormat` function. the payload struct may contains the sequential number, extension name, directory name, mtime, ctime ... etc. but for now we can just put the sequential number in the struct.